### PR TITLE
feat: provide diagnostics for calcs

### PR DESCRIPTION
**Pull request by vetlek**
feat: provide error reports if the calc is not parsable.

feat: provide warnings for calcs that are not supported in Septic.

feat: provide warnings for references to Xvrs that are not in the config.

feat: provide completion for (Sopc)Xvrs. Suggest existing Xvrs when creating new SopcXvr and vice versa. 

feat: provide completion of Xvrs and calcs in CalcPvr.Alg

Release-As: 1.0.0

### CODE REVIEW

**Review by einarsi — CHANGES_REQUESTED** [review]
Looking good! Would be great with a runthrough before approve for merge.

Question: For most linters and syntax checkers, it is possible to either globally disable specific lints or disable them for a section or a single line. 

For Python, setup.cfg can e.g. contain 
```
ignore =
  E203
  W503
```
and lines ending with for instance "# noqa: E501" disables reporting of the line as too long.
The black formatter uses "# fmt: off" and "# fmt: on" to disable autoformatting for sections.

Any way to do something like that here, in case some of the error checks misbehave in certain cases or are annoying for some? Not saying we need it, but worth thinking about how it can be done before implementing lots and lots of syntax checks?

File: server/src/language-service/completionProvider.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+function getRelevantXvrs(
+    curObj: SepticObject,
+    xvrs: SepticObject[]
+): SepticObject[] {
+    let regex;
+    if (["Mvr", "Tvr", "Evr", "Cvr", "Dvr"].includes(curObj.type)) {
+        regex = `^Sopc${curObj.type.charAt(0)}vr`;
+    } else if (
+        ["SopcMvr", "SopcTvr", "SopcCvr", "SopcDvr"].includes(curObj.type)
```
[the remark this answers is not in the document]
Comment (vetlek) [reply server/src/language-service/completionProvider.ts:92]: Fixed

File: server/src/language-service/documentSymbolProvider.ts [hunk 2]
```diff
@@ -134,21 +141,3 @@ function dummySymbol(): DocumentSymbol {
         []
     );
 }
-
-function getSymbolKind(type: string) {
-    if (
-        ["sopcmvr", "sopccvr", "sopctvr", "sopcevr"].includes(
```
[the remark this answers is not in the document]
Comment (vetlek) [reply server/src/language-service/documentSymbolProvider.ts:140]: Delted code, not relevant.

File: package.json [hunk 3]
```diff
@@ -49,7 +49,42 @@
           ],
           "default": "error",
           "description": "Sets the severity level for objects with missing identifier. Ignore implies no validation."
-        },
+        },"septic.diagnostics.alg": {
+					"scope": "window",
+					"type": "string",
+					"enum": [
+						"error",
```
[the remark this answers is not in the document]
Comment (vetlek) [reply package.json:56]: Based on the [documentation](https://code.visualstudio.com/api/references/contribution-points#contributes.configuration) it is not possible to use json $ref in the configuration section, thus not possible to reuse.

**Comment by vetlek** [comment]
[Ref](https://github.com/equinor/vscode-septic/pull/105#pullrequestreview-1404154068): Implementing this on a global scale is easy to do. Marking certain lines is harder to implement and would require some cleaning of the files before using them in SEPTIC (i.e. removing all #...). Think we should be very sure that this is something that is needed before implementing on a local scale.

**Review by einarsi — APPROVED** [review]

**Pull request merged**